deploy/sre-prometheus/bz1980755/100-sre-bz1980755.PrometheusRule.yaml

```
@@ -0,0 +1,21 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  labels:
+    prometheus: k8s
+    role: alert-rules
+  name: sre-bz1980755
+  namespace: openshift-monitoring
+spec:
+  groups:
+  - name: sre-bz1980755
+    rules:
+    - alert: HighSubscriptionSyncRateSRE
+      annotations:
+        message: "The sync rate for the {{ $labels.name }} operator Subscription is high and may indicate it has not installed successfully."
+      expr: |
+        rate(subscription_sync_total{name=~"managed-upgrade-operator|cloud-ingress-operator|osd-metrics-exporter|must-gather-operator|managed-node-metadata-operator|custom-domains-operator|ocm-agent-operator|rbac-permissions-operator|route-monitor-operator|openshift-splunk-forwarder-operator|managed-velero-operator|configure-alertmanager-operator"}[30m]) * 100 > 10
+      for: 5m
+      labels:
+        severity: warning
+        namespace: openshift-monitoring
```

Join subscription_sync_total to catalogsource_ready on name, except strip "-registry" from the name label.  All SRE operators follow this pattern and therefore you can build a generic expression for the alert using exported_namespace.

```suggestion
    rules:
    - expr: subscription_sync_total * on(name) group_right label_replace(catalogsource_ready, "name", "$1", "name", "(.*)-registry")
      record: sre:subscription_sync_total
    - alert: HighSubscriptionSyncRateSRE
      annotations:
        message: "The sync rate for the {{ $labels.name }} operator Subscription is high and may indicate it has not installed successfully."
      expr: |
        rate(sre:subscription_sync_total{exported_namespace=~"openshift.*",exported_namespace!~"openshift-operators.*"}[30m]) * 100 > 10
      for: 5m
      labels:
        severity: warning
        namespace: openshift-monitoring
```

---

Thanks, this is a great suggestion.

I sadly note that three SRE/Hive-distributed operators do not have a catalog that follows the pattern (addon-operator, managed-upgrade-operator, and splunk-forwarder-operator) - they instead end in `-catalog`. However, updating the regex to be `(.*)-(registry|catalog)` was successful in including them without inadvertently including any others, so that's what I've now done.